### PR TITLE
[CLIENT-PY]Optimize array concatenation performance

### PR DESCRIPTION
**Pull request by fuwei3140**
[CLIENT-PY]When fetching data from the server in batches, each np.concatenate() will create a new array and copy the stock array. To optimize performance, after parsing all batches of arrays, execute np.concatenate() only once.

### CODE REVIEW

File: client-py/tests/test_todf.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def test_with_null_query():
         df_output = df_output[df_input.columns.tolist()]
 
         session.close()
-    assert_frame_equal(df_input, df_output)
+    assert_frame_equal(df_input, df_output, check_dtype=False)
```
Comment (SteveYurongSu) [inline client-py/tests/test_todf.py:181]: Why we have to set check_dtype=False?

Comment (HTHou) [inline client-py/tests/test_todf.py:181]:
```suggestion
    assert_frame_equal(df_input, df_output)
```

Comment (HTHou) [reply client-py/tests/test_todf.py]: Fixed

File: client-py/iotdb/utils/IoTDBRpcDataSet.py [hunk 2]
```diff
@@ -266,18 +262,17 @@ def resultset_to_pandas(self):
...
-                    else:
-                        result[column_name] = np.concatenate(
-                            (result[column_name], data_array), axis=0
-                        )
+                result[column_name].append(data_array)
+
         for k, v in result.items():
-            if v is None:
-                result[k] = []
+            if v[0].dtype == "Int32":
```
Comment (HTHou) [inline client-py/iotdb/utils/IoTDBRpcDataSet.py:268]:
```suggestion
            if v is None or len(v) < 1 or v[0] is None:
                result[k] = []
            elif v[0].dtype == "Int32":
```

File: client-py/iotdb/utils/IoTDBRpcDataSet.py [hunk 3]
```diff
@@ -266,18 +262,19 @@ def resultset_to_pandas(self):
...
+            if v is None or len(v) < 1 or v[0] is None:
                 result[k] = []
+            elif v[0].dtype == "Int32":
+                result[k] = pd.Series(np.concatenate(v, axis=0)).astype("Int32")
+            elif v[0].dtype == "Int64":
+                result[k] = pd.Series(np.concatenate(v, axis=0)).astype("Int64")
+            elif v[0].dtype == "boolean":
+                result[k] = pd.Series(np.concatenate(v, axis=0)).astype("boolean")
+            else:
+                result[k] = np.concatenate(v, axis=0)
```
Comment (SteveYurongSu) [inline client-py/iotdb/utils/IoTDBRpcDataSet.py:277]: What about the float number type? Should the float type be considered too?

**Review by SteveYurongSu — APPROVED** [review]
LGTM

**Pull request merged**